corehq/apps/userreports/views.py

```diff
@@ -138,6 +139,7 @@ def create_report(request, domain):
 
 class ReportBuilderView(BaseDomainView):
 
+    @method_decorator(require_permission(Permissions.edit_data))
     @cls_to_view_login_and_domain
```

Comment:
I see now, I believe the `cls_to_view_login_and_domain` checks for user authentication and domain access, without checking what type of user role that web user has in the domain. My addition adds the role check